src/Sylius/Bundle/ApiBundle/Decorator/AuthenticationSuccessHandlerDecorator.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Decorator;
+
+use Sylius\Bundle\ApiBundle\Assigner\CartToUserAssigner;
+use Sylius\Bundle\ApiBundle\Assigner\CartToUserAssignerInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Sylius\Component\User\Model\User;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Http\Authentication\AuthenticationSuccessHandlerInterface;
+
+
+/** @experimental */
+class AuthenticationSuccessHandlerDecorator implements AuthenticationSuccessHandlerInterface
```

Comment:
If I understood correctly, we are putting a cart to session during registration, but we are assigning it to the customer during login. What if after registration someone else will log in? 

I would say, that extending registration or login context, so the token can be sent explicitly to us would be a much better solution. Another option (but I think a little bit worse) would be to have custom cart claiming endpoint